pyqir-generator/pyqir/generator/_native.pyi

```diff
@@ -58,13 +59,104 @@ def const(ty: Type, value: Union[int, float]) -> Value:
 class Builder:
     """An instruction builder."""
 
+    def and_(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a bitwise logical and instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The result.
+        """
+        ...
+
+    def or_(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a bitwise logical or instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The result.
+        """
+        ...
+
+    def xor(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a bitwise logical exclusive or instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The result.
+        """
+        ...
+
+    def add(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts an addition instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The sum.
+        """
+        ...
+
+    def sub(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a subtraction instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The difference.
+        """
+        ...
+
+    def mul(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a multiplication instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The product.
+        """
+        ...
+
+    def shl(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a shift left instruction.
+
+        :param lhs: The value to shift.
+        :param rhs: The number of bits to shift by.
+        :returns: The result.
+        """
+        ...
+
+    def lshr(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a logical (zero fill) shift right instruction.
+
+        :param lhs: The value to shift.
+        :param rhs: The number of bits to shift by.
+        :returns: The result.
+        """
+        ...
+
+    def icmp(self, pred: IntPredicate, lhs: Value, rhs: Value) -> Value:
```

Comment:
Depending on what users are planning to do with `icmp`, we might need to add a branching instruction to the builder, like `BasicQisBuilder.if_result` but conditioned on a boolean instead of a result.

Reply:
I'll do this in a separate PR.